v6.1.11

ned14 opened the pull request:
Merged v6.1.1, the last non-Oracle release.

Split exported libraries into `mysql_shared` and `mysql_static`

caseymcc committed:
fix debug build and windows

caseymcc committed:
add install config

caseymcc committed:
add install config

caseymcc committed:
fix conflict

caseymcc committed:
modified for msvc

caseymcc committed:
fix multi-configurations and ignore gcc warnings

ned14 committed:
Merge upstream v6.1.11.

ned14 committed:
Fixups after rebase of hunter changes on top of merged upstream.

ned14 committed:
Add a static library target to exports in addition to shared library target.

ned14 commented:
@rbsheth A reminder that this PR has been awaiting review for well over a month now. Apologies for pinging you.

rbsheth commented:
@ned14 No worries, you *should* ping me or I won't see it. Looks like there are conflicts?

ned14 commented:
> @ned14 No worries, you _should_ ping me or I won't see it. Looks like there are conflicts?

Oh I didn't know that, sorry. I just assumed you always see all PRs.

I think the conflicts are from the CRLF cleanup and the new `.gitattributes`  to ensure CRLF mismatch doesn't happen again. It's possible the conflicts are instead from the rebase, so I took Ruslan's original delta from trunk, and rebased it onto the last open source release. So that would appear here as a rebase rather than merge i.e. conflict.

ned14 committed:
Fix bad cmake which is causing cmake 3.22 to puke.

ned14 committed:
Add compatibility with externally supplied OpenSSL 1.1.

ned14 committed:
Link to threading library, as if supplying external OpenSSL libraries to use won't link otherwise.

rbsheth commented:
@ned14 Seeing some activity here. Still working on those conflicts?

ned14 commented:
> @ned14 Seeing some activity here. Still working on those conflicts?

Actually no. I have been fixing problems in the work code base. The recent commits do fix useful things needing fixing however.

ned14 committed:
If WITH_SSL is specified to cmake, do not bind in a copy of the SSL libraries
to the MySQL static library as this will produce symbol collisions in anything
consuming the MySQL library and the SSL libraries.

Also fix a build failure in debug builds if WITH_SSL is specified.

ned14 committed:
Update cmake to support use with OpenSSL v3
